relay/src/cli.rs

```diff
@@ -34,13 +32,14 @@ pub fn execute() -> Result<(), Error> {
     // Commands that need a loaded config:
     let mut config = match Config::from_path(config_path) {
         Ok(config) => config,
-        Err(e)
-            if matches.value_of("config").is_none()
-                && e.kind() == ConfigErrorKind::CouldNotOpenFile =>
-        {
-            Config::default()
-        }
-        Err(e) => return Err(e.into()),
+        // TODO(ja): Bring this back
+        // Err(e)
+        //     if matches.value_of("config").is_none()
+        //         && e.kind() == ConfigErrorKind::CouldNotOpenFile =>
+        // {
+        //     Config::default()
+        // }
+        Err(e) => return Err(e),
```

Comment:
So this is how I did it:
```rust
 Err(e) => match e.root_cause().downcast_ref::<ConfigError>() {
     Some(ce)
         if matches.value_of("config").is_none()
             && ce.kind() == ConfigErrorKind::CouldNotOpenFile =>
     {
         Config::default()
     }
     _ => bail!(e),
 },

```

which theoretically should do what it did before, since we know that we are expecting `ConfigError` here.

Reply:
That's maybe even better than what I ended up doing. I opted to move it into a separate function and then return from the innermost branch. 

Note that we can't use `root_cause()` here since that would be some `io` error or something out of our control. Instead, we want to match on the top-most error.